server/sttp-stub-server/src/main/scala/sttp/tapir/server/stub/SttpRequestBody.scala

```diff
@@ -66,4 +76,55 @@ class SttpRequestBody[F[_]](implicit ME: MonadError[F]) extends RequestBody[F, A
     transfer()
     os.toByteArray
   }
+
+  private def extractMultipartParts(parts: Seq[Part[client3.RequestBody[_]]], bodyType: RawBodyType.MultipartBody): List[Part[Any]] = {
+    parts.flatMap { part =>
+      bodyType.partType(part.name).flatMap { partType =>
+        extractPartBody(part, partType).map { body =>
+          Part(
+            name = part.name,
+            body = body,
+            contentType = part.contentType.flatMap(ct => MediaType.parse(ct).toOption),
+            fileName = part.fileName
+          )
+        }
+      }
+    }.toList
+  }
+
+  private def extractPartBody[B](part: Part[client3.RequestBody[_]], bodyType: RawBodyType[B]): Option[Any] = {
+    part.body match {
+      case ByteArrayBody(b, _) =>
+        bodyType match {
+          case RawBodyType.StringBody(charset)  => Some(b)
+          case RawBodyType.ByteArrayBody        => Some(b)
+          case RawBodyType.ByteBufferBody       => Some(ByteBuffer.wrap(b))
+          case RawBodyType.InputStreamBody      => Some(new ByteArrayInputStream(b))
+          case RawBodyType.InputStreamRangeBody => Some(InputStreamRange(() => new ByteArrayInputStream(b)))
+          case RawBodyType.FileBody             => None
```

Comment:
one last suggestion: maybe it would be better to throw an exception here, instead of silently dropping the part? This might be confusing, and an exception will clearly signal that there's a mismatch between the provided & expected part types. Even if it's a limitation of the stub's implementation, it will let the user know what's wrong. (This would need to be combined with a test)